### ISSUE TITLE
support ZSF as snapshot location

### ISSUE THREAD

**Issue by mabod** [teejee2008/timeshift#417]
I want to store the snapshots on a ZFS RAID10 or RAIDZ2. But timeshift is not showing these volumes. Although Issue #291 claims that ZFS is supported: https://github.com/teejee2008/timeshift/issues/291

I am using Manjaro Testing with timeshift 19.01-1

**Comment by defyjoy** [comment]
This has been open for a long long time . Is there update on this ? The above statement is true . IT does not give option neither show the volumes .

**Comment by mabod** [comment]
Yes. Just checked it with timeshift-20.03.r0.gcecd294-1. When I select rsync as the method timeshift is asking me to pick the location of the snapshot and it list the raw devices (my four hard discs) although they are a member of a zfs RAID10. 

Instead of showing the single raw devices - which is really dangerous from my point of view - timeshift should only show the zfs datasets on the RAID10 pool.

**Comment by chudkins** [comment]
I am seeing the same behavior:  My zfs members are all shown, except that only one indicates any free space.  Attached screenshot shows what I mean.

![timeshift_zfs](https://user-images.githubusercontent.com/45431320/88082276-90c66f00-cb4f-11ea-8bfc-024f12e380a7.png)

Expected behavior (or at least, more intuitive) would be to show only the zfs dataset (in my case, "datastore") instead of the raw volumes.  I do see that when I select any of the zfs members shown, the radio buttons for all three are turned on.  Still, it isn't obvious to me as the user what would happen if I tried to proceed.

I am also on Manjaro, with Timeshift package `timeshift-20.03.r8.gad3c01c-1` here.

**Comment by stellarpower** [comment]
I am admittedly using an out-of-date version, although that is precisely why; I need to backup before I upgrade and don't yet have root on ZFS. Timeshift requires (at least on this version) to have a whole ext4 partition and will place its snapshots in the /timshift directory - is there a specific reason for this? I'd far rather be able to point ot an arbitrary directory and use this instead. Currently I am creating an empty ext4 partition large enough for a full backup (a small "trick" volume did not work as it checks it has enough free space) and bind mounting this to my ZFS dataset I use for Timeshift.
